code-challenges.test.js

```diff
@@ -15,35 +15,136 @@
+//I could not figure out how to get these two results in the same function. I did try to make different variables and join the results that way but did not get very far. I was able to get the .sort() method down but it's just calling on the first and last values. I tried manipulating the return portion but was not sure if I had to include a for loop to access those indexes. Brining all the pieces together and writing it correctly is still a struggle for me. 
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const testArray1 = [3, 7, 10, 5, 4, 3, 3]
-const testArray2 = [7, 8, 2, 3, 1, 5, 4]
+// const testArray1 = [3, 7, 10, 5, 4, 3, 3]
+// const testArray2 = [7, 8, 2, 3, 1, 5, 4]
 // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
 
+describe('twoToOneArr', () => {
+    const mergedArray = [3, 7, 10, 5, 4, 8, 2, 1];
+    it('takes in two arrays as arguments and returns one array with no duplicate values', () => {
+        expect([3, 7, 10, 5, 4, 3, 3], [7, 8, 2, 3, 1, 5, 4]).toEqual(expect.arrayContaining(mergedArray));
+    }) 
+    it('does not match if received has duplicate values in the array', () => {
+        expect([3, 3, 3, 3, 7, 7, 5, 5]).not.toEqual(expect.arrayContaining(mergedArray))
+    });
+});
+//Output: Expect takes at most one argument.
 
 // b) Create the function that makes the test pass.
+//PsuedoCode:
+//The arrays need to be put into one array using the spread operator. Even thought there will be duplicate values, it's easier to consolidate both arrays into one at the start. The next step would be to filter through the single merged array and have values that don't repeat be returned in the new array. 
+    //I wanted to use the spread operator because of 1. Trish's trick to remember it and 2. it seemed like it would be a better starting point and just simplify the function. I did have to research further on how to access values and have non-duplicate values being returned.  
+//After figuring out where to put .filter() I looked to the resources to remind me of the .indexOf() method. 
+//Parameters: const testArray1 = [3, 7, 10, 5, 4, 3, 3],const testArray2 = [7, 8, 2, 3, 1, 5, 4]
+//Output: [3, 7, 10, 5, 4, 8, 2, 1]
+
+const testArray1 = [3, 7, 10, 5, 4, 3, 3]
+const testArray2 = [7, 8, 2, 3, 1, 5, 4]
+const mergedTestArrs = [...testArray1, ...testArray2]
+//console.log(mergedTestArrs)   
+const noDups = mergedTestArrs.filter((value, index) => {
+    return mergedTestArrs.indexOf(value) === index;
+});
+console.log(noDups)
+
+//My function ended up working but it is my test that needs to be refactored. The tests were harder to write than the functions for this week's assessments. 😩 Also, for some reason, my tests would be running multiple times in the console.
```

Comment:
Your logic and approach to this problem are great.  I think there are a few things we need to work on with tests and writing functions.  Currently, the logic you have written is living outside of any function which means your test is not truly testing a function.